src/api/java/com/minecolonies/api/colony/buildings/modules/IRegistersBlockModule.java

```diff
@@ -0,0 +1,20 @@
+package com.minecolonies.api.colony.buildings.modules;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Module type to register specific blocks to a building (beds, workstations, etc).
+ */
+public interface IRegistersBlockModule extends IBuildingModule
+{
+    /**
+     * Attempt to register a specific block at a specific module.
+     * @param blockState the state.
+     * @param pos the position.
+     * @param world the world.
+     */
+    void registerBlockPosition(@NotNull BlockState blockState, @NotNull BlockPos pos, @NotNull World world);
+}
```

Comment:
You should add a method that returns all Blocks registered to this module.
Something like `getRegisteredPositions` or `getRegisteredBlocks(final IBlockReader world)`

Reply:
I thought about this, but each building holds the blocks differently for different purposes and sometimes even different sets of blocks. I don't think we need a getter that returns ALL of them